src/Tools/_framework/Paths/CourseActivityEditor.jsx

```diff
@@ -0,0 +1,4647 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
```

Comment:
I did paste this and the portfolio editor into an external diff so I could see what the real new code was, and it looks okay, as discussed offline there is some copy-paste inside the newly added panels that I think could could also be consolidated a bit, but its okay for now.

Taking this approach does have the downside that the test coverage we already have on the portfolio editor isn't covering the common parts between the two, so I'd prefer in the future we try to take the time to share more code as the first approach, even if it will take longer.